assets/src/js/Components/ArchivePicker.js

```diff
@@ -1,151 +1,115 @@
+  const toggleSingleSelection = target => setSelectedIds(isOnlySelected(target) ? [] : [target.id]);
 
-    try {
-      const nextImages = await this.fetchImages( {
-        page: pageIndex,
-        search_text: searchedText,
-      } );
-      const withPageLabel = nextImages.map( image => ( {
-        ...image,
-        pagedTitle: `${ pageIndex } -- ${ image.title }`
-      } ) );
-      this.setState( {
-        currentPage: pageIndex,
-        images: [
-          ...( newSearch ? [] : this.state.images ),
-          ...withPageLabel
-        ],
-        searchText: searchedText,
-      } );
-    } catch ( error ) {
-      this.setState( { error } );
-    } finally {
-      this.setState( { loading: false } );
-    }
-  }
+  const toggleMultiSelection = target => setSelectedIds(
+    selectedIds.includes(target.id)
+      ? selectedIds.filter(id => id !== target.id)
+      : [...selectedIds, target.id]
+  );
 
-  async includeInWp( ids ) {
-    try {
-      this.setState( { processingImages: true } );
-      const updatedImages = await apiFetch( {
-        method: 'POST',
-        path: '/planet4/v1/image-archive/transfer',
-        data: {
-          ids: ids,
-          use_original_language: false,
-        }
-      } );
-      this.updateFromUploadedResponse( updatedImages );
-    } catch ( e ) {
-      console.log( e );
-      this.setState( { processingError: e } );
-    } finally {
-      this.setState( { processingImages: false } );
-    }
-  }
+  useEffect(() => {
+    loadPage();
+  }, []);
 
-  updateFromUploadedResponse( updatedImages ) {
-    const newImages = this.state.images.map( stateImage => {
-      const updated = updatedImages.find( updatedImage => updatedImage.id === stateImage.id );
-      if ( updated ) {
-        return updated;
-      }
-      return stateImage;
-    } );
-    this.setState( {
-      images: newImages,
-    } );
-  }
+  const selectedImages = getSelectedImages(selectedIds);
 
-  async search() {
-    await this.loadNextPage( true );
-  }
-
-  render() {
-    const {
-      loading,
-      error,
-      images,
-    } = this.state;
-
-    return <Fragment>
-      <form
-        className={ 'archive-picker-search' }
-        onSubmit={ event => {
-          event.preventDefault();
-          this.search();
-        } }
-        onChange={ event => this.setState( { enteredSearch: event.target.value } ) }
+  return <Fragment>
+    <form
+      className={'archive-picker-search'}
+      onSubmit={async event => {
+        event.preventDefault();
+        await loadPage(1, enteredSearch);
+        setCurrentPage(1);
+        setSearchText(enteredSearch);
+      }}
+      onChange={event => setEnteredSearch(event.target.value)}
+    >
+      <input
+        type={'text'}
+        disabled={loading}
+      />
+      <input type={'submit'} value={__('Search', 'planet4-master-theme-backend')} />
+    </form>
+    {loading && (
+      <div className={'archive-picker-loading'}> loading...</div>
```

Comment:
There's a [Spinner component](https://github.com/WordPress/gutenberg/tree/162bc50a487f987d1c3250d769f07a28598fe3ae/packages/components/src/spinner) in gutenberg if we want :)

Reply:
I looked into it, we are missing some dependencies to make it work in master-theme... Since this feature is hidden for now and UX needs to be reviewed anyway, we can keep it in mind for then! 🙂